conjure-go-client/httpclient/authn.go

```diff
@@ -69,32 +69,22 @@ func newAuthTokenMiddlewareFromRefreshable(token refreshable.StringPtr) Middlewa
 // (2) an empty BasicAuth and a non-nil error.
 type BasicAuthProvider func(context.Context) (BasicAuth, error)
 
-// basicAuthMiddleware wraps a refreshing BasicAuth pointer and injects basic auth credentials if the pointer is not nil
-type basicAuthMiddleware struct {
-	provider BasicAuthProvider
-}
-
-func (b *basicAuthMiddleware) RoundTrip(req *http.Request, next http.RoundTripper) (*http.Response, error) {
-	basicAuth, err := b.provider(req.Context())
-	if err != nil {
-		return nil, err
-	}
-	setBasicAuth(req.Header, basicAuth.User, basicAuth.Password)
-	return next.RoundTrip(req)
-}
+// BasicAuthOptionalProvider accepts a context and returns either:
+//
+// (1) nil, nil to indicate that no BasicAuth should not be set on the request, or
+//
+// (2) a nonempty BasicAuth and a nil error, or
+//
+// (3) an empty BasicAuth and a non-nil error.
```

Comment:
```suggestion
// (3) a nil BasicAuth and a non-nil error.
```

(in practice looks like it doesn't matter what it returns, since if `err` is non-`nil` the other return value doesn't appear to be inspected, but feels like this should be the recommended/expected behavior)